### PR TITLE
Recalibration database

### PR DESCRIPTION
**Pull request by archiepeterjarvis**
Still needs to be tested on machines

### CODE REVIEW

File: src/asmcnc/comms/router_machine.py [hunk 1]
```diff
@@ -1980,7 +1980,7 @@ def check_SGs_rezero_and_go_to_next_checks_then_tune(self, X = False, Y = False,
 
     def check_temps_and_then_go_to_idle_check_then_tune(self, X = False, Y = False, Z = False):
 
-        if self.motor_driver_temp_in_range(self.s.motor_driver_temp):
+        if self.motor_driver_temp_in_range(self.s.motor_driver_temp) or True:
```
Comment (LettyYeti) [inline src/asmcnc/comms/router_machine.py:1983]: why this change?

File: src/asmcnc/production/database/calibration_database.py [hunk 2]
```diff
@@ -91,9 +92,59 @@ def insert_serial_numbers(self, machine_serial, z_head_serial, lower_beam_serial
...
+            query = "DELETE FROM ZHeadCoefficients WHERE Id = ?"
+
+            cursor.execute(query, combined_id)
+
+            self.delete_coefficients(combined_id)
+
+        self.conn.commit()
+
+    def delete_coefficients(self, combined_id):
+        log("Deleting existing data from Coefficients: " + combined_id)
```
Comment (LettyYeti) [inline src/asmcnc/production/database/calibration_database.py:123]: as above :)

File: src/asmcnc/production/database/test_calibration_database.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+db = calibration_database.CalibrationDatabase()
+db.set_up_connection()
+
+def test_do_z_head_coefficients_exist():
+    assert db.do_z_head_coefficients_exist(167701)
+    assert not db.do_z_head_coefficients_exist(991101)
+
+def test_do_lower_beam_coefficients_exist():
+    assert db.do_lower_beam_coefficients_exist(142926)
+    assert not db.do_lower_beam_coefficients_exist(991101)
```
Comment (LettyYeti) [inline src/asmcnc/production/database/test_calibration_database.py:12]: <3

File: src/asmcnc/production/z_head_qc_jig/z_head_qc_3.py [hunk 4]
```diff
@@ -49,7 +49,7 @@ class ZHeadQC3(Screen):
 
     timer_started = False
     one_minute = 60 # 60 seconds
-    seconds = one_minute*30
+    seconds = one_minute*1
```
Comment (LettyYeti) [inline src/asmcnc/production/z_head_qc_jig/z_head_qc_3.py:52]: ensure this is put back before merging :)

File: src/asmcnc/production/z_head_qc_jig/z_head_qc_3.py [hunk 5]
```diff
@@ -71,7 +71,7 @@ def start_calibration_timer(self):
             self.m.jog_absolute_single_axis('Z', self.m.z_max_jog_abs_limit, 750)
             self.m.jog_relative('X', 30, 6000)
             self.m.jog_relative('X', -30, 6000)
-            self.update_time(30 * self.one_minute) # 30 minutes
+            self.update_time(1 * self.one_minute) # 30 minutes
```
Comment (LettyYeti) [inline src/asmcnc/production/z_head_qc_jig/z_head_qc_3.py:74]: normally I just change the value of one_minute, and then this doesn't need editing either :)

File: src/asmcnc/comms/router_machine.py (hunk 1 above)
Comment (archiepeterjarvis) [reply src/asmcnc/comms/router_machine.py:1983]: force for testing

**Comment by archiepeterjarvis** [comment]
Add the ability to recalibrate components
